message_sender/models.py

```diff
@@ -76,3 +77,28 @@ class OutboundSendFailure(models.Model):
 
     def __str__(self):  # __unicode__ on Python 2
         return str(self.id)
+
+
+@python_2_unicode_compatible
+class Channel(models.Model):
+
+    VUMI_TYPE = 'vumi'
+    JUNEBUG_TYPE = 'junebug'
+
+    CHANNEL_TYPES = (
+        (JUNEBUG_TYPE, 'Junebug'),
+        (VUMI_TYPE, 'Vumi')
+    )
+
+    channel_id = models.CharField(unique=True, null=False, blank=False,
+                                  max_length=64)
+    channel_type = models.CharField(choices=CHANNEL_TYPES, max_length=20,
+                                    default=JUNEBUG_TYPE)
+    concurrency_limit = models.IntegerField(null=False, blank=False, default=0)
+    message_delay = models.IntegerField(null=False, blank=False, default=0)
+    message_timeout = models.IntegerField(null=False, blank=False, default=0)
+    default = models.BooleanField(default=False)
+    configuration = JSONField()
```

Comment:
We might want to include some documentation on this, around what configuration is required for each of the channel types.

We probably also want to validate each configuration against its channel type, so that we cannot save a channel with an invalid configuration.